picard/file.py

```diff
@@ -323,6 +325,38 @@ def _fixed_splitext(self, filename):
             new_filename = ''
         return new_filename, ext
 
+    def _format_filename(self, new_dirname, new_filename, metadata, settings):
+        # TODO: tests !!
+        new_filename, ext = self._fixed_splitext(new_filename)
+        ext = ext.lower()
+        new_filename = new_filename + ext
+
+        # expand the naming format
+        naming_format = settings['file_naming_format']
+        if naming_format:
+            new_filename = self._script_to_filename(naming_format, metadata, settings)
+            # NOTE: the _script_to_filename strips the extension away
+            new_filename = new_filename + ext
+            if not settings['move_files']:
+                new_filename = os.path.basename(new_filename)
+            win_compat = IS_WIN or settings['windows_compatibility']
+            new_filename = make_short_filename(new_dirname, new_filename,
+                                               win_compat)
```

Comment:
Previously we were only passing `settings['windows_compatibility']` here, without the `IS_WIN` check. Looking at `make_short_filename` this mostly relies on the win_compat parameter (except one case were it runs code specific to running on Windows). We should check if the previous behavior was intentional or not. But I think `windows_compatibility` is set forcefully to `True` on Windows, right?

Reply:
Yes, it is on by default on Windows:
https://github.com/metabrainz/picard/blob/master/picard/ui/options/renaming.py#L174-L178

But i still think it's safer to have `IS_WIN or settings['windows_compatibility']` in case of the config file is modified. It's already like that in other places.